### DISCUSSION TITLE
Issue While Configuring Delimiter

### DISCUSSION THREAD

**Discussion by itsexe** [mspnp/AzureNamingTool#98] [Q&A]
Hi Team,

I have been using the tool for a few days and have encountered an issue and have a suggestion to improve its functionality. I attempted to customize the OS disk name by using the VM name as input, so that the tool could generate a disk name with the VM name as a prefix. However, I encountered an issue with the delimiter functionality.

**According to Microsoft [link](https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftcompute), the disk name of a VM supports alphanumerics, underscores, and hyphens. However, when I used a hyphen as a delimiter, the tool did not insert it between the components. On the other hand, when I changed the delimiter to an underscore, the tool generated the name correctly. I would appreciate if you could explain why the hyphen is not being picked up, even though it is a supported character.**

Refer to below 
![hyphen-delimiter](https://github.com/mspnp/AzureNamingTool/assets/62495211/0c68c45c-64a8-4d93-b717-70de633e6678)
![underscore-delimiter](https://github.com/mspnp/AzureNamingTool/assets/62495211/abf10089-ecea-418f-a284-37e4a1d27161)

**Additionally, I have a suggestion to improve the tool's functionality. Currently, the tool only allows the user to choose a single delimiter. However, various resources support both hyphens and underscores, while others only support one or the other. It would be useful if the tool allowed the user to configure multiple delimiters and gave the option to choose different delimiters for different resources.**

**Comment by kislow** [comment]
Release v4.3.0 [Added support for Resource Type editing](https://github.com/mspnp/AzureNamingTool/wiki/Resource-Type-Editing)
